src/tcutility/analysis/vdd/manager.py

```diff
@@ -0,0 +1,220 @@
+﻿from __future__ import annotations
+import pandas as pd
+import numpy as np
+from typing import Dict, List, Optional, Sequence, Union
+import attrs
+import pathlib as pl
+from itertools import zip_longest
+import copy
+import functools
+from tcutility.analysis.vdd import charge
+from tcutility.results import result
+import matplotlib.pyplot as plt
+
+
+AVAILABLE_UNITS_RATIO = {"me": 1000, "e": 1}
+PRINT_FORMAT = {"me": "%+.0f", "e": "%+.3f"}
+
+
+def change_unit_decorator(func):
+    @functools.wraps(func)
+    def wrapper(self, unit: str = "me", *args, **kwargs):
+        current_unit = self.unit
+        self.change_unit(unit)
+        result = func(self, *args, **kwargs)
+        self.change_unit(current_unit)
+        return result
+
+    return wrapper
+
+
+def create_vdd_charge_manager(results: result.Result, name: str = "VDDChargeManager") -> VDDChargeManager:
```

Comment:
Is there a reason to use this function as a constructor rather than VDDChargeManager?
In my opinion it is much cleaner to create these objects via `VDDChargeManager(results)`.

Reply:
I prefer to have the charge manager as a standalone class that takes input of the keys present in the result instance and has not connection with the results instance thereafter. The manager only needs a few variables that are present in the results instance. It is of no use to include the whole results instance in the manager. 

If I would do this in the VDDManager __init__ method, it would couple the results instance to the class which I want to prevent. It could be turned into a classmethod, but I wonder if that makes it cleaner

Reply:
I think the current method would be better than using a class method, lets keep it this way then. I agree that decoupling objects is often a good idea.